### ISSUE TITLE
Bug on doRanging

### ISSUE THREAD

**Issue by AlexisTM** [pozyxLabs/Pozyx-Python-library#47]
The doRanging function on Python is not working as expected.

Indeed there is more likely a **flag not reset** on failure. This leads to a `status == success` on failure and the opposite. Checking the flag **before** calling the function doRanging resets the flag and makes it working. 

Failing code, following the guidelines:

```python
  def measure(self):
    for device in self.devices:
      device_range = pypozyx.DeviceRange()
      status = self.local_device.doRanging(NetworkID(device), device_range, remote_id=None)
      if status == pypozyx.POZYX_SUCCESS:
        print(hex(device), ": ", str(device_range))
      else:
        print hex(device), ": FAILED"
    print " -- "
```

OUTPUT of 0x604F and a wrong ID, sometimes it crashed.

```lisp
device 	| range(mm) 	| strength(dBm)
0x604f : FAILED
0x704b :  657409ms, 167772160mm, 8192dBm
 -- 
0x604f : FAILED
0x704b :  657409ms, 167772160mm, 8192dBm
 -- 
0x604f : FAILED
0x704b :  657409ms, 167772160mm, 8192dBm
```

OUTPUT of 0x604F only AFTER having a wrong ID (**note the FAILED for 1st ranging**)

```lisp
device 	| range(mm) 	| strength(dBm)
0x604f : FAILED
 -- 
0x604f :  101828ms, 4045mm, -92dBm
 -- 
0x604f :  102330ms, 4209mm, -92dBm
 -- 
0x604f :  102833ms, 4181mm, -92dBm
```

Working:

```python
  def measure(self):
    for device in self.devices:
      device_range = pypozyx.DeviceRange()
      status = self.local_device.checkForFlag(POZYX_INT_STATUS_RX_DATA, POZYX_DELAY_INTERRUPT)
      status = self.local_device.doRanging(NetworkID(device), device_range, remote_id=None)
      if status == pypozyx.POZYX_SUCCESS:
        print(hex(device), ": ", str(device_range))
      else:
        print hex(device), ": FAILED"
    print " -- "
```

OUTPUT of the working node with 0x604F and a wrong ID:

```lisp
device 	| range(mm) 	| strength(dBm)
0x604f :  514652ms, 4218mm, -93dBm
0x7070 : FAILED
 -- 
0x604f :  515379ms, 4087mm, -93dBm
0x7070 : FAILED
 -- 
0x604f :  516104ms, 4110mm, -93dBm
0x7070 : FAILED
```

**Comment by AlexisTM** [comment]
The following error happens from time to time on both version:

```python
Traceback (most recent call last):
  File "./range_test.py", line 134, in <module>
    main()
  File "./range_test.py", line 131, in main
    rt.spin()
  File "./range_test.py", line 78, in spin
    self.measure()
  File "./range_test.py", line 101, in measure
    status = self.local_device.doRanging(NetworkID(device), device_range, remote_id=None)
  File "/home/alexis/Desktop/PozyxMultiplatform/pypozyx/lib.py", line 880, in doRanging
    POZYX_DO_RANGING, destination, Data([]), remote_id)
  File "/home/alexis/Desktop/PozyxMultiplatform/pypozyx/core.py", line 287, in useFunction
    return self.regFunction(function, params, data)
  File "/home/alexis/Desktop/PozyxMultiplatform/pypozyx/pozyx_serial.py", line 310, in regFunction
    return int(r[0:2], 16)
ValueError: invalid literal for int() with base 16: ''
```

**Comment by laurentva** [comment]
Hey Alexis,

Thanks for the well-documented issue! I'll see what I can do in fixing this ASAP.

**Comment by laurentva** [comment]
Hey Alexis,

Is it the same device that you upgraded from firmware 1.1 to 1.3 now giving you this unstable serial connection, while it was stable before on firmware version 1.1? I'm investigating whether this might be a firmware issue.

Kind regards,

**Comment by AlexisTM** [comment]
@laurentva I think we are still using the Firmware 1.1 on all devices.

**Comment by laurentva** [comment]
Do you still experience this issue? I can't seem to reproduce it on firmware version 1.3.

**Comment by AlexisTM** [comment]
It might be solved with v1.3, I am using the version with the fix, when I have to use it (not often). 

Let's mark it as closed

**Comment by laurentva** [comment]
Hah, thanks!
